src/storage/src/mut_records.rs

```diff
@@ -98,9 +124,32 @@ impl MutFileRecords {
         if self.f_sink.can_be_appended(buffer.len() as u64) {
             debug!("writing {} bytes at: {}", buffer.len(), self.path.display());
             self.f_sink.write_all(&buffer).await?;
-            // for now, we flush for every send
-            self.f_sink.flush().await?;
-            debug!("flushed {}", self.path.display());
+            match self.flush_policy {
```

Comment:
can you move this code to FlushPolicy? I think this would make it simpler to read and maintain.

Reply:
sure